hedera-node/src/main/java/com/hedera/services/ledger/interceptors/StakingAccountsCommitInterceptor.java

```diff
@@ -373,14 +389,14 @@ private void prepareAuxiliaryArraysFor(final int n) {
 		final var maxImpliedChanges = 3 * n + 1;
 		if (rewardsEarned.length < maxImpliedChanges) {
 			rewardsEarned = new long[maxImpliedChanges];
+			stakeAtStartOfLastRewardedPeriodUpdates = new long[maxImpliedChanges];
 			stakedToMeUpdates = new long[maxImpliedChanges];
-			wasStakeMetaChanged = new boolean[maxImpliedChanges];
 			stakeChangeScenarios = new StakeChangeScenario[maxImpliedChanges];
 			stakePeriodStartUpdates = new long[maxImpliedChanges];
 		}
 		Arrays.fill(rewardsEarned, -1);
```

Comment:
It would be clearer now if we used `Long.MIN_VALUE` for all three sentinel values I think.